controllers/users.js

```diff
@@ -462,6 +464,46 @@ const filterUsers = async (req, res) => {
   }
 };
 
+// one time script function to perform the migration - adding github_user_id field to the document
+const migrate = async (req, res) => {
+  try {
+    // Fetch user data from GitHub API for each document in the users collection
+    // divided by 500 because firestore api guarantee that we can process in batch of 500.
+    const usersSnapshot = await firestore.collection("users").get();
+    const batchCount = Math.ceil(usersSnapshot.docs.length / 500);
+    // Create batch write operations for each batch of documents
+    for (let i = 0; i < batchCount; i++) {
+      const batchDocs = usersSnapshot.docs.slice(i * 500, (i + 1) * 500);
+      const batchWrite = firestore.batch();
+      const batchWrites = [];
+      for (const userDoc of batchDocs) {
+        const githubUsername = userDoc.data().github_id;
+        batchWrite.update(userDoc.ref, { github_user_id: null });
+        batchWrites.push(
+          axios
+            .get(`https://api.github.com/users/${githubUsername}`)
```

Comment:
I have one question: suppose some past user changes there username in github...so there two githubUserName present in rds...during fetch `https://api.github.com/users/${githubUsername}` if they not able to fetch because one of the username data not exist anymore on github...So it will throw error??

Reply:
Nice Question @shubhamsigdar1, Answering that Yes it will throw an error because that ID doesn't exist in the world but it will update all other documents. We need to delete that person's id explicitly. If we don't want to that's completely fine as well. Refer to the attached Video: 

https://github.com/Real-Dev-Squad/website-backend/assets/71075101/113d6d03-5386-43f8-89c4-f5727def4ac1

Reply:
+1, how are we handling the entries which are outdated here?
Like once the GitHub username is changed in that case?

Reply:
As we've discussed. It will add the `github_user_id` field to that document and set it to NULL.